lib/SILGen/SILGenApply.cpp

```diff
@@ -3340,12 +3341,58 @@ void DelayedArgument::emitDefaultArgument(SILGenFunction &SGF,
                                           const DefaultArgumentStorage &info,
                                           SmallVectorImpl<ManagedValue> &args,
                                           size_t &argIndex) {
-  auto value = SGF.emitApplyOfDefaultArgGenerator(info.loc,
-                                                  info.defaultArgsOwner,
-                                                  info.destIndex,
-                                                  info.resultType,
-                                                  info.origResultType);
-  
+  RValue value;  
+  bool isStoredPropertyDefaultArg = false;
+
+  // Check if this is a synthesized memberwise constructor for stored property
+  // default values
+  if (auto ctor = dyn_cast<ConstructorDecl>(info.defaultArgsOwner.getDecl())) {
+    if (ctor->isMemberwiseInitializer() && ctor->isImplicit()) {
+      auto param = ctor->getParameters()->get(info.destIndex);
+      
+      if (auto var = param->getStoredProperty()) {
+        // This is a stored property default arg. Do not emit a call to the
+        // default arg generator, but rather the variable initializer expression
+        isStoredPropertyDefaultArg = true;
+
+        auto pbd = var->getParentPatternBinding();
+        auto entry = pbd->getPatternEntryForVarDecl(var);
+
+        SubstitutionMap subs;
+        auto *genericEnv = ctor->getGenericEnvironmentOfContext();
+        auto typeGenericSig = 
+          var->getDeclContext()->getGenericSignatureOfContext();
+
+        if (genericEnv && typeGenericSig) {
+          subs = SubstitutionMap::get(
```

Comment:
You should be able to say subs = F.getForwardingSubstitutions()

Reply:
Hmm, this might be a bit more complicated than I thought. A simple example like:
```swift
struct X<T> {
  var y: T
  var z = false
}

let x = X(y: 3)
```
The variable initializer that we call in this case is `z`'s whose function signature is:
```sil
$@convention(thin) <T> () -> Bool
```
We can't call this in main because the application uses a generic that isn't a part of main's signature. i.e.
```sil
// function_ref variable initialization expression of X.z
%11 = function_ref @$s4main1XV1zSbvpfi : $@convention(thin) <τ_0_0> () -> Bool // user: %12
%12 = apply %11<T>() : $@convention(thin) <τ_0_0> () -> Bool // user: %14
```
Which crashes during verification because its looking for the `T` generic in main's signature, which there is not. Any ideas?

Reply:
Also, does it make sense that the variable initializer in this case has a generic in its signature? Could this simply be `$@convention(thin) () -> Bool`?